spec/lib/stripe/credit_card_remover_spec.rb

```diff
@@ -4,44 +4,102 @@
 require 'stripe/credit_card_remover'
 
 describe Stripe::CreditCardRemover do
-  let(:credit_card) { double('credit_card', gateway_customer_profile_id: 1) }
+  let!(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
 
-  context 'Stripe customer exists' do
-    context 'and is not deleted' do
-      it 'deletes the credit card clone and the customer' do
-        customer = double('customer', deleted?: false)
-        allow(Stripe::Customer).to receive(:retrieve).and_return(customer)
+  let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
 
-        expect_any_instance_of(Stripe::CreditCardCloneDestroyer).to receive(:destroy_clones).with(
-          credit_card
-        )
-        expect(customer).to receive(:delete)
-        Stripe::CreditCardRemover.new(credit_card).call
-      end
+  let!(:user) { create(:user, email: "apple.customer@example.com") }
+  let!(:enterprise) { create(:enterprise) }
+
+  describe "#remove", :vcr, :stripe_version do
+    before { Stripe.api_key = secret }
```

Comment:
Setting a system config, without later un-setting it makes me uncomfortable (it means the test leaves the system in a different state to when it started).
It looks like you developed the [with_stripe_setup](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/spec/support/request/stripe_helper.rb#L22) helper to do this safely, so I would suggest using that.

But... I'm not sure that's even necessary. Do you know why we have a different environment variable for the test api_key? Why don't we just set the test key in `STRIPE_INSTANCE_SECRET_KEY` for test environments? This will be [loaded automatically](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/config/initializers/stripe.rb#L14) (like it is on production environments).

Reply:
Nice catch. Thanks for this comment.

I've changed the `./config/initializers/stripe.rb` on the last commit to take the test API key for test and CI environments. I think this PR would need to be staged and tested, so we're sure the key is still taken for staging and production environments - right? What do you think @dacook?

Reply:
Thanks, that simplifies things, and clarifies my suggestion a little bit.

My suggestion was to go even further, to remove the variable `STRIPE_SECRET_TEST_API_KEY` altogether. 
We can then add  `STRIPE_INSTANCE_SECRET_KEY` to `.env.test.local`, and add the secret test key there.

Then I think we wouldn't need to change `./config/initializers/stripe.rb` at all, meaning we don't need any manual testing.